src/coreclr/tools/Common/Compiler/HardwareIntrinsicHelpers.cs

```diff
@@ -73,17 +73,13 @@ private static class XArchIntrinsicConstants
             public const int Avx512f = 0x8000;
             public const int Avx512f_vl = 0x10000;
             public const int Avx512bw = 0x20000;
-            public const int Avx512bw_vl = 0x40000;
-            public const int Avx512cd = 0x80000;
-            public const int Avx512cd_vl = 0x100000;
-            public const int Avx512dq = 0x200000;
-            public const int Avx512dq_vl = 0x400000;
-            public const int Avx512Vbmi = 0x800000;
-            public const int Avx512Vbmi_vl = 0x1000000;
-            public const int Serialize = 0x2000000;
-            public const int Avx10v1 = 0x4000000;
-            public const int Avx10v1_v256 = 0x8000000;
-            public const int Avx10v1_v512 = 0x10000000;
+            public const int Avx512cd = 0x40000;
+            public const int Avx512dq = 0x80000;
+            public const int Avx512Vbmi = 0x100000;
+            public const int Serialize = 0x200000;
+            public const int Avx10v1 = 0x400000;
+            public const int Avx10v1_v256 = 0x800000;
+            public const int Avx10v1_v512 = 0x1000000;
```

Comment:
It'd be nice to remove these as well.

The former is unnecessary and the latter is implied by `Avx512F` existing.

Reply:
I can make this change, and just make sure, does this part have conflicting changes with the ongoing Avx10 PR? I know some changes happening there for this as well, but didn't follow closely.

Reply:
That PR is removing `Avx10v1_v256`, so removing it here as well should help avoid a conflict.

Reply:
I saw Avx10v1_V256 was removed from InstructionSetDesc.txt, so what we want here is don't even define `Avx10v1_V256` and `Avx10v1_V512`, and rely on `EVEX` to make sure JIT backend will correctly handle Avx10 nodes? Just to confirm.

Reply:
Right. `Avx10v1_v256` is just part of `Avx10v1`, so it's unnecessary.

Likewise, `Avx10v1_v512` is really just `Avx10v1 + Avx512`, so doing those checks instead fits the need and saves us bits.

Reply:
Thanks for the inputs,

There are duplicated implications there: https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt#L141

Are they intended to be like this, or this shall be removed, I can make the changes if needed.

Reply:
> Are they intended to be like this, or this shall be removed, I can make the changes if needed.

No and I'm actually fixing that in https://github.com/dotnet/runtime/pull/103241

> It looks fine to just remove it from XArchIntrinsicConstants but keep the definition as an ISA, and as mentioned, set InstructionSet_Avx10v1_V512 based on XArchIntrinsicConstants_Avx10v1 and XArchIntrinsicConstants_Avx512f

Right, this is what I had meant should happen.

Any refactoring that changes how `InstructionSet_*` works might be possible but would be a more involved change.

Reply:
Thanks for the explanation, in that case, I can wait for this PR go in first, and I will handle the conflict accordingly.